src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -29,44 +29,185 @@ namespace Castle.Facilities.AspNetCore
 	using Microsoft.AspNetCore.Razor.TagHelpers;
 	using Microsoft.Extensions.DependencyInjection;
 
+	/// <summary>
+	/// For overriding default registration and lifestyles behaviour
+	/// </summary>
+	public class CastleWindsorRegistrationOptions
+	{
+		private readonly IWindsorContainer container;
+		private readonly IServiceCollection services;
+
+		private Assembly entryAssembly = null;
+		private bool controllersAlreadyRegistered = false;
+		private bool tagHelpersAlreadyRegistered = false;
+		private bool viewComponentsAlreadyRegistered = false;
+
+		public CastleWindsorRegistrationOptions(IWindsorContainer container, IServiceCollection services)
+		{
+			this.container = container;
+			this.services = services;
+		}
+
+		internal Assembly EntryAssembly
+		{
+			get
+			{
+				try
+				{
+					entryAssembly = entryAssembly ?? Assembly.GetEntryAssembly();
+				}
+				catch
+				{
+					entryAssembly = entryAssembly ?? Assembly.GetCallingAssembly();
+				}
+
+				return entryAssembly;
+			}
+		}
+
+		internal bool ControllersAlreadyRegistered => controllersAlreadyRegistered;
```

Comment:
Why not just define internal auto properties with a private setter?

Reply:
Yep can do.

Reply:
These are gone now, but yes, the new implementation works this way.